### PR TITLE
`loadAttributesAsTypedArray` now creates packed typed arrays

### PR DESCRIPTION
**Pull request by lilleyse**
Previously there were three different ways to load attributes in GltfLoader: as a WebGL buffer, as a typed array counterpart of the WebGL buffer, and as a packed typed array (de-interleaved)

```
  /**
   * A typed array containing tightly-packed attribute values, as they appear
   * in the model file.
   *
   * @type {Uint8Array|Int8Array|Uint16Array|Int16Array|Uint32Array|Int32Array|Float32Array}
   * @private
   */
  this.packedTypedArray = undefined;
```

```
  /**
   * A typed array containing a CPU copy of the vertex buffer for further
   * processing. Since GPU buffers may have padding, attribute values are
   * accessed using byteOffset and byteStride.
   *
   * @type {Uint8Array}
   * @private
   */
  this.typedArray = undefined;
```

The second version ended up not being very useful and was only complicating the code. Now when attributes are loaded as typed arrays they are always packed. This should work fine for voxels and the new ClassificationModel, which are the only two systems that use the `loadAttributesAsTypedArray` option.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @lilleyse!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/load-as-typed-array-packed/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Source/Scene/GltfLoader.js [hunk 1]
```diff
@@ -698,24 +698,20 @@ function loadAccessor(loader, gltf, accessorId, useQuaternion) {
...
+        accessor,
+        bufferViewTypedArray
+      );
 
-        useQuaternion = defaultValue(useQuaternion, false);
-        loadAccessorValues(accessor, packedTypedArray, values, useQuaternion);
-      })
-      .catch(function () {
-        loadDefaultAccessorValues(accessor, values);
-      });
```
Comment (lilleyse) [inline Source/Scene/GltfLoader.js:718]: It seemed unsafe to catch and ignore errors from the buffer view loader so I removed it. The unit tests still pass so maybe this wasn't actually needed?

File: Source/Scene/GltfLoader.js [hunk 2]
```diff
@@ -883,6 +858,33 @@ function loadAttribute(
...
+      if (loadBuffer) {
+        attribute.buffer = vertexBufferLoader.buffer;
+      }
+
+      if (loadTypedArray) {
+        attribute.typedArray = ComponentDatatype.createArrayBufferView(
+          vertexBufferLoader.quantization.componentDatatype,
+          vertexBufferLoader.typedArray.buffer
+        );
+      }
```
Comment (lilleyse) [inline Source/Scene/GltfLoader.js:871]: This refactor exposed a bug related to packed typed arrays of Draco content. Basically `getPackedTypedArray` is not quantization aware and it fails with an out of bounds error.

Here Draco uses its own method to create the typed array.

File: Source/Scene/GltfLoader.js [hunk 3]
```diff
@@ -979,8 +986,23 @@ function loadInstancedAttribute(
...
+  // - translations are required for 2D
+  // - GPU instancing is not supported.
+  let loadTypedArray =
+    loader._loadAttributesAsTypedArray ||
+    ((hasRotation || !hasTranslationMinMax) && isTransformAttribute) ||
+    modelSemantic === InstanceAttributeSemantic.FEATURE_ID ||
+    !frameState.context.instancedArrays;
+
+  const loadBuffer = !loadTypedArray;
+  loadTypedArray = loadTypedArray || loadFor2D;
```
Comment (lilleyse) [inline Source/Scene/GltfLoader.js:1005]: I tried to put all the load-as-typed-array logic in one place. Previously it was spread across this function and the calling code.

File: Source/Scene/GltfVertexBufferLoader.js [hunk 4]
```diff
@@ -384,7 +384,11 @@ function loadFromDraco(vertexBufferLoader) {
     }
 
     // Now wait for process() to run to finish loading
-    vertexBufferLoader._typedArray = typedArray;
+    vertexBufferLoader._typedArray = new Uint8Array(
+      typedArray.buffer,
+      typedArray.byteOffset,
+      typedArray.byteLength
+    );
```
Comment (lilleyse) [inline Source/Scene/GltfVertexBufferLoader.js:391]: Before this fix `_typedArray` was not actually a `Uint8Array`.

This and https://github.com/CesiumGS/cesium/pull/10541/files#r919520142 were causing Draco attributes to fail when packed.

File: Source/Scene/GltfLoader.js (hunk 2 above)
Comment (ptrgags) [inline Source/Scene/GltfLoader.js:871]: Could we factor out a function to contain this logic? this section of code is getting rather nested.

File: Source/Scene/GltfLoader.js [hunk 5]
```diff
@@ -883,6 +858,33 @@ function loadAttribute(
...
+        // The accessor's byteOffset and byteStride should be ignored since values
+        // are tightly packed in a typed array
+        const bufferViewTypedArray = vertexBufferLoader.typedArray;
+        attribute.typedArray = getPackedTypedArray(
+          gltf,
+          accessor,
+          bufferViewTypedArray
+        );
+        attribute.byteOffset = 0;
+        attribute.byteStride = undefined;
```
Comment (ptrgags) [inline Source/Scene/GltfLoader.js:886]: similarly here, factor out a function

**Review by ptrgags — COMMENTED** [review]
@lilleyse changes are working nicely! I only had a small suggestion about refactoring a function to limit nested ifs

**Comment by lilleyse** [comment]
@ptrgags updated

**Comment by ptrgags** [comment]
This looks good now, thanks @lilleyse!

**Pull request merged**